glengine/gl_engine.h

```diff
@@ -12,16 +12,26 @@
 namespace glengine {
 
 struct State;
+class Context;
 
 class GLEngine {
+  public:
+    struct Config {
+        // debug flags
+        bool show_framebuffer_texture = false;
+        bool show_imgui_statistics = false;
+        uint16_t msaa_samples = 4;
+        bool use_mrt = false;
+    };
+
   public:
     virtual ~GLEngine();
 
     // ////// //
     // engine //
     // ////// //
 
-    bool init(const Config &config = {});
+    bool init(Context *context, const Config &config);
```

Comment:
this is the main change on the user side: the engine has to be explicitly initialized with a window/rendering context